### DISCUSSION TITLE
ESP32-C3 SuperMini Not Connecting

### DISCUSSION THREAD

**Discussion by lukewakeford** [lathoub/Arduino-BLE-MIDI#89] [Q&A] [answered]
I'm having issues getting anything to connect to the ESP32-C3 SuperMini.

I have a basic sketch working and I can see the BLE device from my iPad running OnSong, the device is also discoverable on my mac via the Midi Studio -> Bluetooth configuration but does not let me connect.

Possible I'm missing something obvious, any help appreciated.

My sketch:
```
#include <BLEMIDI_Transport.h>
#include <hardware/BLEMIDI_ESP32_NimBLE.h>

BLEMIDI_CREATE_INSTANCE("BluetoothMidi", MIDI)

void setup()
{
  pinMode(LED_BUILTIN, OUTPUT);
  digitalWrite(LED_BUILTIN, HIGH);

  BLEMIDI.setHandleConnected(OnConnected);
  BLEMIDI.setHandleDisconnected(OnDisconnected);
   
  MIDI.begin();
}

void loop()
{
  MIDI.read();
}

void OnConnected() {
  digitalWrite(LED_BUILTIN, LOW);
}

void OnDisconnected() {
  digitalWrite(LED_BUILTIN, HIGH);
}
```

|iPad Connecting|iPad Not Connected|Mac|
|---|---|---|
|![Connecting](https://github.com/lathoub/Arduino-BLE-MIDI/assets/1557818/10351115-169f-4299-99da-6f40cc295830)|![NotConnected](https://github.com/lathoub/Arduino-BLE-MIDI/assets/1557818/ab2cbeb2-2d90-4155-8cd4-a63d006eb457)|![Mac Bluetooth](https://github.com/lathoub/Arduino-BLE-MIDI/assets/1557818/23d3f4ac-254f-4dad-9f91-65aeb9ee42ea)|

**Comment by lathoub** [comment]
Hi,
Can you get the basic BLE stuff working from the underlying lib (see examples)? https://github.com/h2zero/NimBLE-Arduino
THe ESP32C3 is indicated as supported by NimBLE-Arduino

**Reply by lukewakeford** [reply]
Hey thanks for the reply, I have just tried a few of the examples in the NimBLE-Arduino library.

- The `NimBLE_Scan` seems to work as I can see other BT devices appearing and disappearing in the console.
- The `NimBLE_Server` behaves the same as above, the device is advertising as `NimBLE-Arduino` but I am unable to connect to it.

I did spot someone talking about 'extended advertising' and the requirement to enable `CONFIG_BT_NIMBLE_EXT_ADV` in `nimconfig.h` for the c3, s3 and h2 - I tried this out with the extended advertising example but this resulted in no device being advertised at all.

i'll raise an issue in that repo and see where we get to 👍🏻 thanks.

**Reply by lukewakeford** [reply, marked as answer]
Well, after all that, I stumbled upon the same issue over [here](https://github.com/h2zero/NimBLE-Arduino/discussions/674) Seems like the [answer](https://github.com/h2zero/NimBLE-Arduino/discussions/674#discussioncomment-9656255) is that some boards have bad xtal accuracy?

I've now tested the `NimBLE_Server` example with another board (from the same manufacturer) and it's allowed me to connect. 🤦🏻
